### PR TITLE
DAH-1134 finish additional info section

### PR DESCRIPTION
**Pull request by jacksongonzales**
This PR:
- Adds text truncation
  - After some exploration, I found that the approach necessary for our weird use-case (of truncating a string containing HTML tags before injecting the string into raw HTML) was going to be overcomplicated to reconcile with the Bloom component. Basically, the Bloom component works well for a string that is in its final state, i.e. not going to be parsed as a separate type of entity, which is what we need to do with the Salesforce data we're pulling.

### CODE REVIEW

File: app/javascript/components/TextTruncate.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          onClick={() => setIsExpanded(!isExpanded)}
+          onKeyPress={() => setIsExpanded(!isExpanded)}
+          role="button"
+          tabIndex={0}
+        >
+          {isExpanded ? t("label.less") : t("label.more")}
+        </span>
+      )}
+    </div>
+  )
```
Comment (christine-sfg) [inline app/javascript/components/TextTruncate.tsx:38]: Is it possible to add a unit test for this?

**Comment by christine-sfg** [comment]
@jacksongonzales It's working for 'Special Notes' which has alot of text, but for 'Important Program Rules' there is a 'More' link, but no additional text displays after clicking.

image

**Comment by christine-sfg** [comment]
'Important Program Rules' is fixed now. No longer seeing a 'More' link on that section

File: app/javascript/__tests__/components/TextTruncate.test.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    const { getByText } = await renderAndLoadAsync(<TextTruncate text={content} />)
+
+    // check for presence of an ellipsis
+    getByText(/\.\.\./)
+  })
```
Comment (christine-sfg) [inline app/javascript/__tests__/components/TextTruncate.test.tsx:13]: Thanks for adding the test! Another scenario to test would be to make sure it doesn't unnecessarily truncate.

**Review by christine-sfg — APPROVED** [review]

File: app/javascript/__tests__/components/TextTruncate.test.tsx (hunk 2 above)
Comment (jacksongonzales) [reply app/javascript/__tests__/components/TextTruncate.test.tsx:13]: weirdly this is the scenario that was hard to configure 😭 maybe we can work on getting that working another time

**Pull request merged**